Update SHOW STATUS docs (#15363)
pingcap/docs: status-variables.md

---
title: Server Status Variables
summary: Use status variables to see the system and session status
---

# Server Status Variables

> Most status variables are implemented to be compatible with MySQL. These variables give information about the global status of the server and the status of the current session.

Reviewer: ```suggestion
Server status variables provide information about the global status of the server and the status of the current session in TiDB. Most of these variables are designed to be compatible with MySQL.
```

> The [SHOW GLOBAL STATUS](/sql-statements/sql-statement-show-status.md) command can be used to retrieve the global status and the [SHOW SESSION STATUS](/sql-statements/sql-statement-show-status.md) command can be used to see the status of the current session.
> 
> The [FLUSH STATUS](/sql-statements/sql-statement-flush-status.md) command is also support to aid MySQL compatibility.

Reviewer: ```suggestion
You can retrieve the global status using the [SHOW GLOBAL STATUS](/sql-statements/sql-statement-show-status.md) command, and the status of the current session using the [SHOW SESSION STATUS](/sql-statements/sql-statement-show-status.md) command. 

Additionally, the [FLUSH STATUS](/sql-statements/sql-statement-flush-status.md) command is supported for MySQL compatibility.
```

> ## Variable Reference

Reviewer: ```suggestion
## Variable reference
```

Author: `system-variables.md` is also using `Variable Reference`. Are they both wrong?

Reviewer: Yes, they are both wrong here. According to our style guide, we use sentence case for all headings except for H1.

Author: Fixed the other one in 37a8e95c3938e2068db927c955fb1a43600bb0ad

> ### Compression

Reviewer: All the variables are read-only, right?

- Scope: SESSION
- Type: Boolean
- Indicates if the MySQL Protocol uses compression or not.

### Compression_algorithm

- Scope: SESSION
- Type: String
- Indicates the compression algorithm that is used for the MySQL Protocol.

### Compression_level

> - Scope: SESSION | GLOBAL

Reviewer 2: I do not see `Compression_level` from ` SHOW GLOBAL STATUS;`

```
% mysql --comments --host 127.0.0.1 --port 4000 -u root --compress
...
mysql> select tidb_version()\G
*************************** 1. row ***************************
tidb_version(): Release Version: v7.6.0-alpha
Edition: Community
Git Commit Hash: 3f94666b4bafd903886b89206ac603d26d9a9e87
Git Branch: heads/refs/tags/v7.6.0-alpha
UTC Build Time: 2023-11-18 14:25:32
GoVersion: go1.21.3
Race Enabled: false
Check Table Before Drop: false
Store: tikv
1 row in set (0.01 sec)
+-----------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------->
| Variable_name         | Value                                                                                                                                                                                                                                              >
+-----------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------->
| Ssl_cipher            | TLS_AES_128_GCM_SHA256                                                                                                                                                                                                                             >
| Ssl_cipher_list       | RC4-SHA:DES-CBC3-SHA:AES128-SHA:AES256-SHA:AES128-SHA256:AES128-GCM-SHA256:AES256-GCM-SHA384:ECDHE-ECDSA-RC4-SHA:ECDHE-ECDSA-AES128-SHA:ECDHE-ECDSA-AES256-SHA:ECDHE-RSA-RC4-SHA:ECDHE-RSA-DES-CBC3-SHA:ECDHE-RSA-AES128-SHA:ECDHE-RSA-AES256-SHA:E>
| Ssl_server_not_after  | Feb 18 01:23:56 2024 UTC                                                                                                                                                                                                                           >
| Ssl_server_not_before | Nov 20 01:23:56 2023 UTC                                                                                                                                                                                                                           >
| Ssl_verify_mode       | 5                                                                                                                                                                                                                                                  >
| Ssl_version           | TLSv1.3                                                                                                                                                                                                                                            >
| Uptime                | 2592                                                                                                                                                                                                                                               >
| ddl_schema_version    | 50                                                                                                                                                                                                                                                 >
| server_id             | 04ec893d-7eea-4abb-91f5-a572799f4e7f                                                                                                                                                                                                               >
| tidb_gc_last_run_time | 20231120-11:05:56.379 +0900                                                                                                                                                                                                                        >
| tidb_gc_leader_desc   | host:yahondambp.local, pid:11983, start at 2023-11-20 10:23:56.187046 +0900 JST m=+14.986920604                                                                                                                                                    >
| tidb_gc_leader_lease  | 20231120-11:08:56.389 +0900                                                                                                                                                                                                                        >
| tidb_gc_leader_uuid   | 62fa94ed2240001                                                                                                                                                                                                                                    >
| tidb_gc_safe_point    | 20231120-10:55:56.379 +0900                                                                                                                                                                                                                        >
+-----------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------->
(END)
```

Reviewer 2: I'm wondering if it should be `Scope: SESSION`.

Author: Yes correct. I've updated the scope in 5820f655ff976d8e4d9b7324d211133f651e8160 and verified that there are 9 in global scope and 13 in session scope.

```
sql> \pager tail -1
Pager has been set to 'tail -1'.

sql> SHOW GLOBAL STATUS;
9 rows in set (0.0059 sec)

sql> SHOW SESSION STATUS;
13 rows in set (0.0055 sec)

sql> \pager
Pager has been disabled.

sql> SELECT VERSION();
+------------------------------------------+
| VERSION()                                |
+------------------------------------------+
| 8.0.11-TiDB-v7.6.0-alpha-223-g3f94666b4b |
+------------------------------------------+
1 row in set (0.0014 sec)
```

```
$ grep -E 'Scope:.*GLOBAL' status-variables.md | wc -l
9
$ grep -E 'Scope:.*SESSION' status-variables.md | wc -l
13
```

- Type: Integer
- The compression level that is used for the MySQL Protocol.

### Ssl_cipher

- Scope: SESSION | GLOBAL
- Type: String
- TLS Cipher that is in use.

### Ssl_cipher_list

- Scope: SESSION | GLOBAL
- Type: String
- The list of TLS Ciphers that the server supports.

### Ssl_server_not_after

- Scope: SESSION | GLOBAL
- Type: Date
- The expiration date of the X.509 certificate of the server that is used for TLS connections.

### Ssl_server_not_before

- Scope: SESSION | GLOBAL
- Type: String
- The start date of the X.509 certificate of the server that is used for TLS connections.

### Ssl_verify_mode

- Scope: SESSION | GLOBAL
- Type: Integer
- The TLS verification mode bitmask.

### Ssl_version

- Scope: SESSION | GLOBAL
- Type: String
- The version of the TLS protocol that is used

### Uptime

- Scope: SESSION | GLOBAL
- Type: Integer
- Uptime of the server in seconds.

### ddl_schema_version

- Scope: SESSION | GLOBAL
- Type: Integer
- The version of the DDL schema that is used.

### last_plan_binding_update_time <span class="version-mark">New in v5.2.0</span>

> - Scope: SESSION | GLOBAL

Reviewer 2: `last_plan_binding_update_time` only appears at `show session status`.

Reviewer 2: I'm wondering if it should be `Scope: SESSION`

Author: Yes correct. I have fixed this in 5820f655ff976d8e4d9b7324d211133f651e8160

- Type: Timestamp
- The time and date of the last plan binding update.

### server_id

- Scope: SESSION | GLOBAL
- Type: String

> - UUID of the server

Reviewer: ```suggestion
- The UUID of the server.
```
